src/helpers/ClauseResultsHelpers.ts

```diff
@@ -347,6 +352,78 @@ export function findLocalIdsForComparisonOperators(
   }
 }
 
+/**
+ * Helper function to kick off the recursive search for the relationship (With, Without) source's localId in annotation
+ * structure. If found this returns the localId of the parent node of the given source localId. Null is returned if not
+ * found.
+ */
+function findRelationshipAliasAnnotationId(annotation: Annotation[], sourceLocalId: string): string | null {
+  for (const a of annotation) {
+    const id = findRelationshipAliasNodeAnnotationId([a.s], sourceLocalId);
+    if (id) {
+      return id;
+    }
+  }
+  return null;
+}
+
+/**
+ * Recursively looks through the annotation structure for the source localId and grab its parent id which also contains
+ * the alias. This search is valid to be used with relationship (With, Without) clauses and  allows us to tag the found
+ * localId, which only exists in the annotation, as the alias localId for the source so it can be highlighted when the
+ * source clause has a result.
+ *
+ * For example in the below snippet. We are looking for the source localId 355 and want to return 301 which includes
+ * the alias.
+ *
+ *  {
+ *    "r": "301",
+ *    "s": [
+ *      {
+ *        "r": "355",
+ *        "s": [
+ *          {
+ *            "s": [
+ *              {
+ *                "value": [
+ *                  "\"Bladder Cancer Diagnosis\""
+ *                ]
+ *              }
+ *            ]
+ *          }
+ *        ]
+ *      },
+ *      {
+ *        "value": [
+ *          " ",
+ *          "BladderCancer"
+ *        ]
+ *      }
+ *    ]
+ *  }
+ *
+ * @returns id of the node that is parent to the source localId if found
+ */
+function findRelationshipAliasNodeAnnotationId(
+  annotation: AnnotationStatement[],
+  sourceLocalId: string
+): string | null {
+  for (const as of annotation) {
```

Comment:
Suggest a different iterator variable than `as` which may be confusing since "as" can be a key word in some contexts.